dbt/redshift/redshift_etl.py

```diff
@@ -0,0 +1,123 @@
+        parsely_redshift.copy_from_s3(network=network,
+                                s3_prefix=prefix,
+                                table_name=table_name,
+                                host=host,
+                                user=user,
+                                password=password,
+                                database=database,
+                                port=port,
+                                access_key_id=access_key_id,
+                                secret_access_key=secret_access_key)
+
+    # This runs dbt once all of the new data has been copied into the raw data table
+    dpl_wd = os.path.join(os.getcwd(), 'dbt/redshift/')
+    subprocess.call(dpl_wd + "run_parsely_dpl.sh " + dbt_profiles_dir + ' ' + dbt_target, shell=True, cwd=dpl_wd)
+
+
+def main():
+    parser = parsely_redshift.get_default_parser("Amazon Redshift utilities for Parse.ly")
+    parser.add_argument('--start_date', required=False, default=ETL_START_DATE,
+                        help='The first day to process data from S3 to Redshift in the format YYYY-MM-DD')
+    parser.add_argument('--end_date', required=False, default=ETL_END_DATE,
+                        help='The last day to process data from S3 to Redshift in the format YYYY-MM-DD')
+    parser.add_argument('--dbt_profiles_dir', required=False, default=DBT_PROFILE_LOCATION,
+                        help='The location from root that contains the .dbt/profiles.yml file, example: /home/user/.dbt/')
+    parser.add_argument('--dbt_target', required=False, default=DBT_PROFILE_TARGET_NAME,
+                        help='The target ie. dev, prod, or test to use within the dbt profiles.yml file.')
+    parser.add_argument('--create-table', action='store_true', default=True,
+                        help='Optional: create the Redshift Parse.ly rawdata table because it does not yet exist.')
+    args = parser.parse_args()
+
+    # Reset dbt_profile to any updated settings:
+    settings_migration = migrate_settings()
+    if not settings_migration:
+        logging.warning("Settings not copied to dbt_profiles.yml successfully.")
+
+    # Handle defaults
+    if args.create_table:
+        try:
+            parsely_redshift.create_table(
+                table_name=get_settings_arg_mapping_value('table_name', args.table_name),
+                host=get_settings_arg_mapping_value('host', args.redshift_host),
+                user=get_settings_arg_mapping_value('user', args.redshift_user),
+                password=get_settings_arg_mapping_value('password', args.redshift_password),
+                database=get_settings_arg_mapping_value('database', args.redshift_database),
+                port=get_settings_arg_mapping_value('port', args.redshift_port),
+                keep_extra_data=get_settings_arg_mapping_value('keep_extra_data', args.keep_extra_data)
+            )
+        except psycopg2.Error:
+            logging.warning("Table already exists, skipping create table statement.")
```

Comment:
Same question here. I'm not too familiar with how dbt works, so maybe it would raise an exception further down anyway - but it would be good to warn the user and exit early, so they can address the issue.

Reply:
I should make this a logging instead of a warning, it's by design. Thank you for seeing that